nck/utils/date_handler.py

```diff
@@ -40,18 +40,12 @@ def __get_previous_month_dates(current_date: date) -> Tuple[date, date]:
 
 
 def check_date_range_definition_conformity(start_date: date, end_date: date, date_range: str):
-    if (start_date is None and end_date is None) and date_range is None:
-        raise NoDateDefinitionException(
-            "You must at least define a couple \
-                        start-date/end-date or a date-range"
-        )
-    elif (start_date is None or end_date is None) and (start_date is not None or end_date is not None):
-        raise MissingDateDefinitionException("If you define dates, both start_date and end_date must be defined")
-    if start_date is not None and end_date is not None and date_range is not None:
-        raise InconsistentDateDefinitionException(
-            "You must define either start_date and end_date \
-            or date_range, but not both"
-        )
+    if not any([start_date, end_date, date_range]):
+        raise NoDateDefinitionException("You must at least define a couple (start-date, end-date) or a date-range")
+    elif bool(start_date) != bool(end_date):
+        raise MissingDateDefinitionException("Both start_date and end_date must be defined")
+    elif all([start_date, end_date, date_range]):
+        raise InconsistentDateDefinitionException("You must define either (start_date, end_date) or date_range but not both")
```

Comment:
Why did we define various exceptions here?
A simple DateDefinitionException is not enough?

Reply:
I agree, exception names are very close in this case, and don't allow us to make a real distinction between the different cases, so a single `DateDefinitionException` should be enough.

Reply:
Shouldn't we also throw an exception if `start_date` > `end_date`?

Reply:
Probably yes because in this case, the native exception thrown by the API is :
"The date range you've entered is invalid."